test/powershell/Modules/Microsoft.PowerShell.Core/Get-Module.Tests.ps1

```diff
@@ -99,4 +99,53 @@ Describe "Get-Module" -Tags "CI" {
         $modules.Name -join "," | Should -BeExactly "Bar,Foo,Foo,Zoo,Zoo"
         $modules[3].Path | Should -BeExactly (Resolve-Path "$testdrive\Modules\Zoo\Too\Zoo.psm1").Path
     }
+
+    It "Get-Module -FullyQualifiedName <FullyQualifiedName> -ListAvailable" {
+        $moduleSpecification  = @{ModuleName = "Foo"; ModuleVersion = "2.0"}
+        $modules = Get-Module -FullyQualifiedName $moduleSpecification -ListAvailable
+        $modules | Should -HaveCount 1
+        $modules.Name | Should -BeExactly "Foo"
+        $modules.Version | Should -Be "2.0"
+    }
+
+    It "Get-Module <Name> -Refresh -ListAvailable" {
+        $modules = Get-Module -Name 'Zoo' -ListAvailable
+        $modules | Should -HaveCount 1
+        $modules.Name | Should -BeExactly "Zoo"
+        $modules.ExportedFunctions.Count | Should -Be 0 -Because 'No exports were defined'
+
+        New-ModuleManifest -Path "$testdrive\Modules\Zoo\Zoo.psd1" -FunctionsToExport 'Test-ZooFunction'
+
+        $modules = Get-Module -Name 'Zoo' -ListAvailable -Refresh
+        $modules | Should -HaveCount 1
+        $modules.Name | Should -BeExactly "Zoo"
+        $modules.ExportedFunctions.Count | Should -Be 1 -Because 'We added a new function to export'
```

Comment:
```suggestion
        $modules.ExportedFunctions | Should -HaveCount 1 -Because 'We added a new function to export'
```

Reply:
Not for this one. This is a generic dictionary and `-HaveCount` incorrectly returns 1 no matter how many functions there are.